### ISSUE TITLE
Segmentation fault

### ISSUE THREAD

**Issue by 1311894932** [NVlabs/CALM#11]
i was running ase and calm on wsl2, and the output is :
------------------------------------------------------------------------------------------------------------------------------------------------
Importing module 'gym_37' (/home/zhouyang/task_python/isaacgym/python/isaacgym/_bindings/linux-x86_64/gym_37.so)
Setting GYM_USD_PLUG_INFO_PATH to /home/zhouyang/task_python/isaacgym/python/isaacgym/_bindings/linux-x86_64/usd/plugInfo.json
PyTorch version 1.13.1+cu117
Device count 1
/home/zhouyang/task_python/isaacgym/python/isaacgym/_bindings/src/gymtorch
Using /home/zhouyang/.cache/torch_extensions/py37_cu117 as PyTorch extensions root...
Emitting ninja build file /home/zhouyang/.cache/torch_extensions/py37_cu117/gymtorch/build.ninja...
Building extension module gymtorch...
Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
ninja: no work to do.
Loading extension module gymtorch...
2023-09-11 22:38:29,256 - INFO - logger - logger initialized
Error: FBX library failed to load - importing FBX data will not succeed. Message: No module named 'fbx'
FBX tools must be installed from https://help.autodesk.com/view/FBX/2020/ENU/?guid=FBX_Developer_Help_scripting_with_python_fbx_installing_python_fbx_html
MOVING MOTION DATA TO GPU, USING CACHE: True
Importing module 'rlgpu_37' (/home/zhouyang/task_python/isaacgym/python/isaacgym/_bindings/linux-x86_64/rlgpu_37.so)
Setting seed: 9815
Started to train
Not connected to PVD
+++ Using GPU PhysX
/buildAgent/work/99bede84aa0a52c2/source/gpucommon/src/PxgCudaMemoryAllocator.cpp (59) : warning : Failed to allocate pinned memory.

/buildAgent/work/99bede84aa0a52c2/source/gpucommon/src/PxgCudaMemoryAllocator.cpp (59) : warning : Failed to allocate pinned memory.

/buildAgent/work/99bede84aa0a52c2/source/gpucommon/src/PxgCudaMemoryAllocator.cpp (59) : warning : Failed to allocate pinned memory.

/buildAgent/work/99bede84aa0a52c2/source/gpucommon/src/PxgCudaMemoryAllocator.cpp (59) : warning : Failed to allocate pinned memory.

Physics Engine: PhysX
Physics Device: cuda:0
GPU Pipeline: enabled
Cylinder is not natively supported, tesellated mesh will be used
/buildAgent/work/99bede84aa0a52c2/source/gpucommon/include/PxgCudaUtils.h (54) : internal error : SynchronizeStreams cuEventRecord failed

/buildAgent/work/99bede84aa0a52c2/source/gpucommon/include/PxgCudaUtils.h (60) : internal error : SynchronizeStreams cuStreamWaitEvent failed

/buildAgent/work/99bede84aa0a52c2/source/gpunarrowphase/src/PxgNarrowphaseCore.cpp (2408) : internal error : memcpy failed fail!
  700

/buildAgent/work/99bede84aa0a52c2/source/gpucommon/src/PxgCudaMemoryAllocator.cpp (59) : warning : Failed to allocate pinned memory.

Segmentation fault

Does anyone know anything about this bug? Any good suggestions? I would appreciate it!

**Comment by 1311894932** [comment]
actually i run calm first, and succeed...but when i succesfully run deepmimic, i meet this bug in ase and calm(they use same env), i was so confused

**Comment by 1311894932** [comment]
allright...it may be the isaac gym, i run the joint_monkey.py and still segmentation fault

**Comment by hhebb** [comment]
i used wsl2 long day ago, segmentation fault error made often.
so i recommend to install ubuntu. or dual booting.

**Comment by 1311894932** [comment]
> i used wsl2 long day ago, segmentation fault error made often. so i recommend to install ubuntu. or dual booting.

thank u so much!! maybe i should try dual booting...but i have a question, have you used isaacgym for some days?because i'm new in RL, and i've never encountered this kind of bug before

**Comment by hhebb** [comment]
i think that's not about isaacgym error.
but it's wsl error. i think

**Comment by 1311894932** [comment]
> i think that's not about isaacgym error. but it's wsl error. i think

okay...thanks again for your reply

**Comment by 1311894932** [comment]
solve it by installing ubuntu
